src/main/java/com/example/labj1/logic/OrderLogic.java

```diff
@@ -27,26 +24,30 @@ public class OrderLogic {
 
 
     @Transactional
-    public void createOrder(Order order) {
+    public ResponseEntity<?> createOrder(Order order) {
         System.out.println("order: " + order);
         String inventoryUrl = "http://localhost:8081/inventory/v2";
 
         UriComponentsBuilder url = UriComponentsBuilder.fromUriString(inventoryUrl)
 
                 .queryParam("productId", (order.getLineItems().get(0).getProductInfo().getId()));
-        // .queryParam("productId", 24234);
-
         HttpHeaders headers = new HttpHeaders();
         headers.setAccept(Collections.singletonList(MediaType.APPLICATION_JSON));
 
         HttpEntity<String> entity = new HttpEntity<>(headers);
         Inventory inventory = restTemplate.exchange(url.build().toUri().toString(), HttpMethod.GET, entity, Inventory.class).getBody();
         System.out.println("inventory data :: " + inventory.getTotalQuantity());
         if (order.getLineItems().get(0).getProductInfo().getId() == inventory.getProductId() && order.getLineItems().get(0).getQuantity() <= inventory.getTotalQuantity()) {
-            System.out.print("true");
```

Comment:
Add a for loop so you can handle multiple products in the same order. Only persist/place the order after all products are proven "available".

If all items are available persist the order and maybe return nothing (void) or return an object with details about the outcome of the save. I'd start with void and only add things as you need them. We may want to return the original order just to show what got persisted.

Throw an exception if an item is not available in inventory.  Here's an article on exception handling in Spring Boot. https://reflectoring.io/spring-boot-exception-handling/